spec/server/functional/products.spec.coffee

```diff
@@ -6,6 +6,28 @@ request = require '../request'
 moment = require 'moment'
 libUtils = require '../../../server/lib/utils'
 
+describe 'GET /db/products', ->
+  beforeEach utils.wrap ->
+    yield utils.clearModels([User, Payment])
+    yield utils.populateProducts()
+    @user = yield utils.initUser()
+    yield utils.loginUser(@user)
+
+  it 'shouldnt leak coupon code information', utils.wrap ->
+      url = utils.getURL('/db/products/')
+      [res, body] = yield request.getAsync({url})
+      doc = JSON.parse(body)
```

Comment:
You can have the `body` parsed by doing this: `yield request.getAsync({url, json:true})`